tests/particle_tracer/test_particle.py

```diff
@@ -97,3 +98,52 @@ class TestPhysics:
     """
     Test suite for the Physics class.
     """
+
+class TestTime:
+    """
+    Test suite for the Time class.
+    """
+
+    def test_initial_current_time(self):
+        """
+        Test that the initial current time matches the reference date.
+        """
+        reference_date = np.datetime64('2023-01-01T00:00:00', 's')
+        offset_seconds = np.timedelta64(0, 's')
+        time_instance = Time(reference_date=reference_date, offset_seconds=offset_seconds)
+
+        assert time_instance.get_current_time() == reference_date, \
+            "Initial current time should match the reference date."
+
+    def test_update_add_seconds(self):
+        """
+        Test updating the current time by adding seconds.
+        """
+        reference_date = np.datetime64('2023-01-01T00:00:00', 's')
+        offset_seconds = np.timedelta64(0, 's')
+        time_instance = Time(reference_date=reference_date, offset_seconds=offset_seconds)
+
+        delta_seconds = np.timedelta64(120, 's')
+        time_instance.update(delta_seconds)
+
+        expected_time = reference_date + delta_seconds
+        assert time_instance.get_current_time() == expected_time, \
+            f"Updated current time should be {expected_time}, but got {time_instance.get_current_time()}."
+
+    def test_update_subtract_seconds(self):
```

Comment:
simulation time will move very likely only forwards. But it is good check it can be use either way.